src/xpk/core/storage.py

```diff
@@ -0,0 +1,340 @@
+        self.auto_mount,
+        self.mount_point,
+        self.readonly,
+        self.manifest,
+    ]
+
+  def _get_bucket(self) -> str:
+    """
+    Retrieves the bucket name from PersistentVolume definition associated with the storage.
+
+    Returns:
+        The name of the bucket.
+    """
+    client = k8s_client.CoreV1Api()
+    try:
+      pv: V1PersistentVolume = client.read_persistent_volume(self.pv)
+    except client.ApiException as e:
+      xpk_print(
+          f"Exception when calling CoreV1Api->read_persistent_volume: {e}"
+      )
+    return pv.spec.csi.volume_handle
+
+  def get_mount_options(self) -> list[str]:
+    """
+    Retrieves the mount options for the PersistentVolume.
+
+    Returns:
+        A list of mount options.
+    """
+    client = k8s_client.CoreV1Api()
+    try:
+      pv: V1PersistentVolume = client.read_persistent_volume(self.pv)
+    except client.ApiException as e:
+      xpk_print(
+          f"Exception when calling CoreV1Api->read_persistent_volume: {e}"
+      )
+    return pv.spec.mount_options
+
+
+def list_storages(k8s_api_client: ApiClient) -> list[Storage]:
+  """
+  Lists all Storage custom resources in the cluster.
+
+  Args:
+      k8s_api_client: An ApiClient object for interacting with the Kubernetes API.
+
+  Returns:
+      A list of Storage objects representing the Storage resources.
+  """
+  api_instance = k8s_client.CustomObjectsApi(k8s_api_client)
+  try:
+    resp = api_instance.list_cluster_custom_object(
+        group=XPK_API_GROUP_NAME,
+        version=XPK_API_GROUP_VERSION,
+        plural=STORAGE_CRD_KIND.lower() + "s",
+    )
+  except ApiException as e:
+    xpk_print(f"Kubernetes API exception while listing Storages: {e}")
+    xpk_exit(1)
+
+  storages = []
+  for stg in resp["items"]:
+    storage = Storage(stg)
+    storages.append(storage)
+  return storages
+
+
+def get_auto_mount_storages(k8s_api_client: ApiClient) -> list[Storage]:
+  """
+  Retrieves all Storage resources that have --auto-mount flag set to true.
+
+  Args:
+      k8s_api_client: An ApiClient object for interacting with the Kubernetes API.
+
+  Returns:
+      A list of Storage objects that have `auto_mount` set to True.
+  """
+  auto_mount_storages: list[Storage] = []
+  for storage in list_storages(k8s_api_client):
+    if storage.auto_mount is True:
+      auto_mount_storages.append(storage)
+  return auto_mount_storages
+
+
+def get_storages(k8s_api_client: ApiClient, names: list[str]) -> list[Storage]:
+  """
+  Retrieves a list of Storage resources by their names, including auto-mounted storages.
+
+  Args:
+      k8s_api_client: An ApiClient object for interacting with the Kubernetes API.
+      names: A list of Storage resource names to retrieve.
+
+  Returns:
+      A list of Storage objects matching the given names and any auto-mounted storages.
+  """
+  storages: list[Storage] = []
+  for storage in list_storages(k8s_api_client):
+    if storage.name in names:
```

Comment:
If "names" is the list of storages requested by users to mount to the workload, here we are ignoring the ones that are not in the list_storages. This can be a problematic behavior because user might think all the storage they requested are mounted.

We need to throw an error if any requested storage doesn't exist and let the user know the name of invalid storages. Maybe there is a typo there and they can fix it.

Reply:
Good point, thanks. Done!